src/modules/list.js

```diff
@@ -0,0 +1,111 @@
+import Task from './task.js';
+
+export class List {
+  tasks = []
+
+  workForm = null;
+
+  constructor() {
+    this.container = document.getElementById('form-item');
+
+    const clearCompletedBtn = document.querySelector('.list-footer');
+    clearCompletedBtn.onclick = this.clearCompleted.bind(this);
+
+    if (localStorage.getItem('Task')) {
+      this.tasks = JSON.parse(localStorage.getItem('Task')).map((task) => new Task(task.description, task.completed, task.index));
+    }
+
+    this.setCurrentForm();
+  }
+
+  setCurrentForm(name = 'addTaskForm', isUpdating = false) {
+    this.workForm = new TaskForm(name, isUpdating);
+    this.workForm.form.onsubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    if (this.workForm.isUpdating) {
+      this.updateTask(this.workForm.form.description.value, this.workForm.form.index.value);
+    } else {
+      this.addTask(this.workForm.form.description.value);
+      this.workForm.form.description.focus();
+      this.workForm.form.reset();
+    }
+  }
+
+  addTask(description) {
+    const task = new Task(description, false, this.tasks.length + 1);
+    this.tasks.push(task);
+    this.render().saveToLocal();
+  }
+
+  updateTask(description, index) {
+    this.tasks[index - 1].description = description;
+    this.saveToLocal();
+  }
+
+  removeTask(index) {
+    this.tasks = this.tasks.filter((task) => task.index !== index);
+    for (let i = 0; i < this.tasks.length; i += 1) {
+      this.tasks[i].index = i + 1;
+    }
+    this.render().saveToLocal();
+  }
+
+  toggleTaskStatus(task) {
+    task.toggleStatus();
+    this.render().saveToLocal();
+  }
+
+  clearCompleted() {
+    this.tasks = this.tasks.filter((task) => !task.completed);
+    this.render().saveToLocal();
+  }
+
+  render() {
+    this.container.parentNode.querySelectorAll('[data-task]').forEach((task) => task.remove());
+
+    this.tasks.sort((a, b) => b.index - a.index).forEach((task) => {
+      const { taskNode, descriptionNode, taskIndex } = task.createNode();
+      this.container.after(taskNode);
+      descriptionNode.onfocus = (e) => this.editing(e, taskNode, taskIndex);
+      descriptionNode.onblur = (e) => this.edited(e, taskNode);
+      const completeBtn = taskNode.querySelector('input[type="checkbox"]');
+      completeBtn.onclick = () => this.toggleTaskStatus(task);
+    });
+
+    this.tasks.sort((a, b) => a.index - b.index);
+
+    return this;
+  }
+
+  saveToLocal() {
+    localStorage.setItem('Task', JSON.stringify(this.tasks));
+  }
+
+  editing(e, taskNode, taskIndex) {
+    taskNode.classList.add('editing-task');
+    const trash = taskNode.querySelector('button.trash');
+    trash.classList.toggle('hidden');
+    this.setCurrentForm(`task-${taskIndex}`, true);
+    trash.onmousedown = () => {
+      this.removeTask(taskIndex);
+      this.setCurrentForm();
+    };
+    taskNode.querySelector('i.fa-bars').classList.toggle('hidden');
+  }
+
+  edited(e, taskNode) {
```

Comment:
Your JS codes has a good structure based on use of classes and javaScript modules. I used your app, but whenever I edit the description of todo list, at the end I press enter and nothings happens. You need to add an key event for pressing enter. while it is pressed it should change the property of input tag. search about input readonly and input type text.
🧑‍💼  keep coding!!